From 75edd76994a75417c3fc51888adedd8d3ae99889 Mon Sep 17 00:00:00 2001
From: Manisha Agarwal <maniagar@codeaurora.org>
Date: Thu, 2 Apr 2020 15:22:48 +0530
Subject: [PATCH 36/38] Audio: Update current active bt device of
 mApmConnectedDevices.

Update key of bt-a2dp inside mApmConnectedDevices when the
active bt device changes.

CRs-Fixed: 2634767
Change-Id: I92fa39581a10fd8900e41139d71ca46dd2445eb6
---
 .../core/java/com/android/server/audio/AudioDeviceInventory.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/core/java/com/android/server/audio/AudioDeviceInventory.java b/services/core/java/com/android/server/audio/AudioDeviceInventory.java
index adbcb68d..d5388199 100644
--- a/services/core/java/com/android/server/audio/AudioDeviceInventory.java
+++ b/services/core/java/com/android/server/audio/AudioDeviceInventory.java
@@ -459,6 +459,7 @@ public class AudioDeviceInventory {
 
             if (event == BtHelper.EVENT_ACTIVE_DEVICE_CHANGE) {
                 // Device is connected
+                mApmConnectedDevices.replace(AudioSystem.DEVICE_OUT_BLUETOOTH_A2DP, key);
                 if (a2dpVolume != -1) {
                     mDeviceBroker.postSetVolumeIndexOnDevice(AudioSystem.STREAM_MUSIC,
                             // convert index to internal representation in VolumeStreamState
-- 
2.28.0

